plugins/blockly/blocks/util/apply_first_match.ts

```diff
@@ -2,29 +2,126 @@ import Blockly from "blockly";
 
 Blockly.Blocks["apply_first_match"] = {
   init: function () {
-    this.appendDummyInput().appendField("Apply first match");
-    this.appendValueInput("ACTIONS_ONE").setCheck("Action[]");
-    this.appendValueInput("ACTIONS_TWO").setCheck("Action[]");
-    this.setOutput(true, "Action[]");
+    this.appendDummyInput().appendField("apply first match");
+    this.appendValueInput("ACTIONS0").setCheck("Action[]");
+    this.setPreviousStatement(true);
+    this.setNextStatement(true);
     this.setColour(230);
     this.setTooltip("");
     this.setHelpUrl("");
+
+    this.setMutator(new Blockly.Mutator(["additional_action"]));
+    this.addtional_num_actions = 0;
```

Comment:
type & naming convention
```suggestion
    this.additionalNumActions = 0;
```